new-frontiers-14/frontier-station-14
# Halloween: Cauldron to replace deep fryers

## About the PR
* Adds a new entity, the **cauldron**, which is functionally identical to a deep fryer
* Adds a new `HalloweenSystem` that effectively migrates all deep fryers to cauldrons, replacing them on map creation

## Why / Balance
Spooky witch stuff! Brew things, or something.

The many identical-looking fill levels for the cauldron were added for compatibility with the deep fryer. I could set it to having only a single fill level, but then you can't even see if there's anything in the cauldron until it's full, which isn't ideal. This solution also isn't ideal, but it's for a single day only, so.

The cauldron can be deconstructed and reconstructed to make a normal deep fryer. This is intentional: (1) it gives players a way to "opt out" of the funny meme, (2) I'm too lazy to go and define a unique board for this once-a-year playfulness.

## How to test
1. Either set your computer's clock to 31st of October or modify the Halloween holiday to be active on whatever day you're testing this.
2. Spawn on the dev map. The deep fryer should now be a cauldron.
3. Warp to Tinnia's. The deep fryer should now be a cauldron.
4. Warp to Bahama Mama's. The deep fryer should now be a cauldron.
5. Buy a food ship. Its deep fryer should now be a cauldron.
6. Use the cauldron like a normal deep fryer.
7. Deconstruct and reconstruct the cauldron to get a deep fryer.
8. Ensure that the cauldrons on POIs block tool use.
9. Try doing deep dark magic with the cauldron. You know, for science.

## Media
https://github.com/user-attachments/assets/910c8987-fffb-4656-8ba6-0a5aea2e3a70

## Requirements

## Breaking changes
No.

**Changelog**
N/A, let people discover as they play the game.
